image-segmentation/custom-recipes/score_image/recipe.py

```diff
@@ -0,0 +1,140 @@
+
+# Local path to trained weights file
+model_folder_input = get_input_names_for_role('model_folder')[0]
+model_folder_path = dataiku.Folder(model_folder_input).get_path()
+COCO_MODEL_PATH = os.path.join(model_folder_path, "mask_rcnn_coco.h5")
+# Download COCO trained weights from Releases if needed
+
+#Directory of images to run detection on
+image_folder_input = get_input_names_for_role('image_folder')[0]
+image_folder = dataiku.Folder(image_folder_input)
+IMAGE_DIR = image_folder.get_path()
+
+image_folder_output = get_output_names_for_role('scored_folder')[0]
+images_segmented = dataiku.Folder(image_folder_output)
+SEGMENTED_IM_DIR = images_segmented.get_path()
+
+details_folder_name = get_output_names_for_role('details_folder')[0]
+details_folder = dataiku.Folder(details_folder_name)
+DETAILS_DIR = details_folder.get_path()
+# -------------------------------------------------------------------------------- NOTEBOOK-CELL: CODE
+class InferenceConfig(coco.CocoConfig):
+    # Set batch size to 1 since we'll be running inference on
+    # one image at a time. Batch size = GPU_COUNT * IMAGES_PER_GPU
+    GPU_COUNT = 1
+    IMAGES_PER_GPU = len(image_folder.list_paths_in_partition(partition=''))
+
+config = InferenceConfig()
+
+model = modellib.MaskRCNN(mode="inference", model_dir=model_folder_path, config=config)
+model.load_weights(COCO_MODEL_PATH, by_name=True)
+
+
+class_names = ['BG', 'person', 'bicycle', 'car', 'motorcycle', 'airplane',
+               'bus', 'train', 'truck', 'boat', 'traffic light',
+               'fire hydrant', 'stop sign', 'parking meter', 'bench', 'bird',
+               'cat', 'dog', 'horse', 'sheep', 'cow', 'elephant', 'bear',
+               'zebra', 'giraffe', 'backpack', 'umbrella', 'handbag', 'tie',
+               'suitcase', 'frisbee', 'skis', 'snowboard', 'sports ball',
+               'kite', 'baseball bat', 'baseball glove', 'skateboard',
+               'surfboard', 'tennis racket', 'bottle', 'wine glass', 'cup',
+               'fork', 'knife', 'spoon', 'bowl', 'banana', 'apple',
+               'sandwich', 'orange', 'broccoli', 'carrot', 'hot dog', 'pizza',
+               'donut', 'cake', 'chair', 'couch', 'potted plant', 'bed',
+               'dining table', 'toilet', 'tv', 'laptop', 'mouse', 'remote',
+               'keyboard', 'cell phone', 'microwave', 'oven', 'toaster',
+               'sink', 'refrigerator', 'book', 'clock', 'vase', 'scissors',
+               'teddy bear', 'hair drier', 'toothbrush']
+
+# -------------------------------------------------------------------------------- NOTEBOOK-CELL: CODE
+to_score = image_folder.list_paths_in_partition(partition='')
+image_list=[]
+for im in to_score :
+    image_path = IMAGE_DIR+im
+    image = skimage.io.imread(image_path)
+    image_list.append(image)
+
+result=model.detect(image_list, verbose=1)
+
+for i, im in enumerate(result):
+    result[i]["image_path"] = to_score[i]
+result_json=[]
+
+def jsonify(data):
```

Comment:
to improve: put all classes/functions inside python lib

Reply:
done, in utils.py